bigquery/indexer.py

```diff
@@ -150,6 +150,25 @@ def _field_docs_by_id(id_prefix, name_prefix, fields):
             yield field_id, field_dict
 
 
+# Sample and participant tables need to be indexed differently.
+# For participant tables, we can use partial updates
+# (https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html#_updates_with_a_partial_document)
+#
+# If one participant table has weight and another has height:
+# - First weight table is indexed. Participant documents get a weight field.
+# - Then height table is indexed. Participant documents get a height field. The
+#   weight fields are unchanged.
+#
+# Now say one sample table has center and another has platform.
+# - First center table is indexed. For each participant document, a samples
+#   field is created. The samples field contains an array of nested objects,
+#   each of which has a center field.
+# - Then platform table is indexed. For each participant document, the samples
```

Comment:
nit: maybe add that it's only for rows with the same value for the `sample_id_column` between multiple tables that would get overwritten.

Reply:
Discussed offline. It's actually participant_id that matters, not sample_id. If two rows from two tables have the same participant_id and different sample_id, the samples field will still be overwritten.